src/executor/vm/jit.rs

```diff
@@ -301,6 +311,23 @@ impl JitPrototype {
             Ok(result.get(0).map(|v| TryFrom::try_from(v).unwrap()))
         });
 
+        // Zero the memory if necessary.
+        if !self.memory_zeroed {
+            if let Some(memory) = self.memory.as_ref() {
+                let offset = cmp::min(
+                    memory.data_size(),
+                    usize::try_from(zero_memory_above).unwrap(),
+                );
+
+                // Soundness: the documentation of wasmtime precisely explains what is safe or not.
+                // Basically, we are safe as long as we are sure that we don't potentially grow the
+                // buffer (which would invalidate the buffer pointer).
+                unsafe {
+                    memory.data_unchecked_mut()[offset..].fill(0);
+                }
```

Comment:
don't we want to set `self.memory_zeroed` to `true` here?

Reply:
`self` is thrown away afterwards so this doesn't matter